drivers/vmwarevsphere/flags.go

```diff
@@ -278,3 +295,23 @@ func (f *FileAttrFlag) SetPerms(owner, group, perms int) {
 func (f *FileAttrFlag) Attr() types.BaseGuestFileAttributes {
 	return &f.GuestPosixFileAttributes
 }
+
+func (d *Driver) SetMachineOSFromFlags(flags drivers.DriverOptions) error {
+	switch {
+	case flags.String("--vmwarevsphere-os") == "":
+		d.OS = defaultMachineOS
+		return nil
+	default:
+		d.OS = strings.ToLower(flags.String("--vmwarevsphere-os"))
+		err := d.CheckMachineOS(flags.String(d.OS))
+		if !err {
+			return fmt.Errorf("[SetMachineOSFromFlags] Machine [%s] has an unsupported MachineOS [%s]\n", d.MachineName, d.OS)
+		}
+		return nil
+	}
```

Comment:
I switched to using an if and brought the same error along from before. This ensures that we don't pass a bad OS value beyond this point to fail early instead of when the cloud-config is being parsed.